addons/mail/static/src/core/common/failure_model.js

```diff
@@ -65,6 +65,11 @@ export class Failure extends Record {
     }
 
     get body() {
+        if (this.notifications.length === 1) {
+            return _t('An error occurred when sending an email on "%(record_name)s"', {
+                record_name: this.lastMessage?.thread.name,
+            });
+        }
```

Comment:
We're usually checking existence of thread (in the rest of the file we do).
Also it doesn't make sense to return this value if there's no last message, it should not use optional chaining but move to condition.